src/main/java/com/intuit/graphql/orchestrator/schema/transform/FieldResolverTransformerPostMerge.java

```diff
@@ -118,10 +121,28 @@ private FieldResolverContext updateWithTargetFieldData(final FieldResolverContex
     List<ResolverArgumentDefinition> updatedResolverArgumentDefinitions = updateWithInputType(
             resolverDirectiveDefinition.getArguments(), targetFieldInputValueDefinitions, fieldResolverContext);
 
+    FieldContext targetFieldContext = new FieldContext(getParentTypeName(targetFieldDefinition), targetFieldDefinition.getName());
+    AtomicReference<String> targetNamespaceRef = new AtomicReference<>();
+
+    //checking if resolver is for a nested field
+    if(resolverDirectiveDefinition.getField().contains(".")) {
+      //using desc which contains namespace for the parent type
+      String parentDesc = sourceXtextGraph.getType(getParentTypeName(targetFieldDefinition)).getDesc();
```

Comment:
@CNAChino recently added code to traverse `if(resolverDirectiveDefinition.getField().contains(".")))`  because of a bug, That will give u access to the namespace or coderegistry, you should reuse that instead of getting it from description.

Reply:
for FQN based and non-FQN resolver field, we can use FieldContext(getParentTypeName(targetFieldDefinition), targetFieldDefinition.getName()) to access the coderegistry

Reply:
but in case of nested service and it is only the single nested service, there's a need to traverse to root field get the entry from codeRegistry